app/recipes/routes.py

```diff
@@ -0,0 +1,422 @@
+# ------------------------------------------- #
+
+# ----- CREATE ----- #
+@recipes.route("/desserts/new", methods=["GET", "POST"])
+def desserts_new():
+    """
+    CRUD - Create recipe for database.
+
+    Inject all form data to new recipe document on submit.
+    """
+    if request.method == "GET":
+        allergen_list = dropdown_allergens()
+        dessert_list = dropdown_dessert_type()
+        measurement_list = dropdown_measurement()
+        return render_template(
+            "desserts_new.html",
+            allergens=allergen_list,
+            desserts=dessert_list,
+            measurements=measurement_list)
+
+    if request.method == "POST":
+        # get today's date and date recipe was last edited
+        today = datetime.now().strftime("%d %B, %Y")
+        last_edit = int(datetime.now().strftime("%Y%m%d"))
+        # get user / author details
+        session_user = get_user_lower(session["user"])["username"]
+        author = users_collection.find_one({"username": session_user})["_id"]
+        # get and convert total time
+        hours = int(request.form.get("total_hrs")) * 60 if request.form.get("total_hrs") != "" else ""
+        total_time = int(request.form.get("total_mins")) + hours if hours != "" else int(request.form.get("total_mins"))
+        # get form data prior to submitting
+        submit = {
+            "recipe_name": request.form.get("recipe_name"),
+            "recipe_slug": slugify(request.form.get("recipe_name")),
+            "description": request.form.get("description"),
+            "dessert_type": request.form.get("dessert_type"),
+            "ingredient_amount": request.form.getlist("ingredient_amount"),
+            "ingredient_measurement": request.form.getlist("ingredient_measurement"),
+            "ingredient_name": request.form.getlist("ingredient_name"),
+            "directions": request.form.getlist("directions"),
+            "total_hrs": request.form.get("total_hrs"),
+            "total_mins": request.form.get("total_mins"),
+            "total_time": total_time,
+            "allergens": request.form.getlist("allergens"),
+            "img_src": request.form.get("img_src"),
+            "author": author,
+            "date_added": today,
+            "date_updated": today,
+            "last_edit": last_edit,
+            "views": 0,
+            "user_favs": 0
+        }
+        # get the new _id being created on submit
+        newID = recipes_collection.insert_one(submit)
+        # add recipe _id to user's recipe list
+        users_collection.update_one({"_id": ObjectId(author)}, {"$push": {"user_recipes": newID.inserted_id}})
+        # slugify url to be user-friendly
+        slugUrl = slugify(request.form.get("recipe_name"))
+        flash(Markup(f"<i class='far fa-check-circle green-text'></i> Sounds delicious! Thanks for adding this recipe!"))
+        # if selected, add recipe to user-favs as well
+        if request.form.get("add_favs") == "on":

```

Comment:
Don't use "on" and "off" for `add_favs` (or anything that can have only one of 2 values that checks a state and doesn't need an actual value). Use a truthy/falsy value, say 0 and 1 or true and false. Then at this point, you can simply do
```
if request.form.get("add_favs"):
```